presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -1414,6 +1414,24 @@ else if (firstException != exception) {
         throw propagate(firstException);
     }
 
+    // TODO instead of whitelisting exceptions we propagate we should recognize exceptions which we suppress and try different alternative call
+    // this requires product tests with HDP 3
+    private static boolean isValidExceptionalResponse(Exception exception)
+    {
+        requireNonNull(exception, "exception is null");
```

Comment:
I don't think adding this for a private utility method is helpful. We know that the only caller doesn't pass null.